packages/peregrine/lib/talons/CategoryTree/useCategoryLeaf.js

```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react';
+
+/**
+ * Returns props necessary to render a CategoryLeaf component.
+ *
+ * @param {object} props
+ * @param {function} props.onNavigate - callback to fire on link click
+ * @return {{ handleClick: function }}
+ */
+export const useCategoryLeaf = props => {
+    const { onNavigate } = props;
+
+    const handleClick = useCallback(() => {
+        onNavigate();
```

Comment:
This isn't doing anything but wrapping/renaming the function. I think we don't need this talon.

Reply:
In general, a callback received as a prop shouldn't get directly applied as an event handler. I think it's important to wrap the event handler so that `event` doesn't get passed to `onNavigate`; `onNavigate` should be called with no arguments.

Reply:
OK -- in that case keep your eye out on my PRs cause I think I did not wrap these callbacks in some of my talons.